test/lib/validators/content_view_puppet_module_validator_test.rb

```diff
@@ -5,52 +5,83 @@
 module Katello
   class ContentViewPuppetModuleValidatorTest < ActiveSupport::TestCase
     def setup
+      @rhel_repository = katello_repositories(:rhel_6_x86_64)
       @repository = katello_repositories(:p_forge)
-      @base_record = { :errors => { :base => [] }, :content_view => OpenStruct.new(:puppet_repos => [@repository]) }
-      @validator = Validators::ContentViewPuppetModuleValidator.new(:attributes => [:name])
+      @base_record = { errors: { base: [] }, content_view: OpenStruct.new(puppet_repos: [@repository], organization: @repository.organization) }
+      @validator = Validators::ContentViewPuppetModuleValidator.new(attributes: [:name])
     end
 
-    test "fails if both name and uuid blank" do
-      @model = OpenStruct.new(:errors => {:base => []})
+    test "passes if name and author match a puppet module" do
+      # fixture katello_puppet_modules(:abrt)
+      @model = OpenStruct.new(@base_record.merge(name: "abrt", author: "johndoe"))
+      @validator.validate(@model)
+      assert_empty @model.errors[:base]
+    end
+
+    test "fails if only name provided" do
+      @model = OpenStruct.new(@base_record.merge(name: "abrt"))
       @validator.validate(@model)
 
       refute_empty @model.errors[:base]
     end
 
-    test "fails if only name provided" do
-      @model = OpenStruct.new(@base_record.merge(:name => "module name"))
+    test "fails if only author provided" do
+      @model = OpenStruct.new(@base_record.merge(author: "johndoe"))
       @validator.validate(@model)
 
       refute_empty @model.errors[:base]
     end
 
-    test "passes if name and author provided" do
-      puppet = PuppetModule.create!(:name => "module name", :author => "module author", :uuid => '9932943299423')
-      puppet.repositories << @repository
-      @model = OpenStruct.new(@base_record.merge(:name => "module name", :author => "module author"))
+    test "fails if name and author do not match a puppet module" do
+      # fixture katello_puppet_modules(:abrt)
+      @model = OpenStruct.new(@base_record.merge(name: "abrt", author: "janedoe"))
       @validator.validate(@model)
-      assert_empty @model.errors[:base]
+      refute_empty @model.errors[:base]
+    end
+
+    test "fails if both name and uuid blank" do
+      @model = OpenStruct.new(errors: {base: []})
+      @validator.validate(@model)
+
+      refute_empty @model.errors[:base]
     end
 
-    test "passes if uuid provided" do
-      @model = OpenStruct.new(@base_record.merge(:uuid => "3bd47a52-0847-42b5-90ff-206307b48b22"))
+    test "passes if uuid matches a puppet module" do
+      # fixture katello_puppet_modules(:abrt)
```

Comment:
What about grabbing that from the fixture so as not to hard code the UUID and avoid comments that are confusing?